integration/networktest/functional/native_token.go

```diff
@@ -0,0 +1,81 @@
+package functional
```

Comment:
I still find the abstractions a bit confusing. I think there's some subtle overlap in responsibilities, making it a bit hard to reason about them.

I'll try to outline my understanding.

- `runner`: Entry point, where you setup and wire together the other components
- `traffic`: Something that just generates traffic against a network, but does not mess with the network.
- `functional` - I can't think of any meaningful difference from the "traffic" stuff.
- `scenarios` - where you can mess with the nodes.


Would something like this be possible?
In the entry point do the following:

1. set up the network you want to use. (I think the abstractions for this bit are right)

2. set up the network failure profile ( most of the the stuff that's in scenarios right now)
(this could provide some hooks: `onNodeStopped`, `onNodeStarted`). This only works against the local networks

3. set up the functional stuff (stuff that is in `traffic` and `functional`). 

4. Wire them together.

I'm thinking that to keep things decoupled there could be a standard lifecycle channel exposed where the network and the network failure and maybe something else could push events.

This channel is then injected everywhere so components can make decisions based on what's going on in the environment